### PR TITLE
Trust region code cleanup

### LINKED ISSUE #16748: Clean up the code for the trust region minimizers.
First write unit tests for MoreSorensen and DTRS minimizers, to be abel to test that your changes do not affect the operation of the classes.

- Make the code of `DTRSMinimizer` and `MoreSorensenMinimizer` conform to the mantid coding standards (function and variable names). 
- Remove unused `nlls_inform` structure. Replace error code setting with exceptions.
- Make members of `NLLS_workspace` struct private or protected members of the minimizer classes. Remove `NLLS_workspace`.

### PR DESCRIPTION
**Pull request by KarlPalmen**
Add unit tests for DTRS and MoreSorensen fitting function minimizers and tidy the code as described in the issue being fixed.

**To test:**

Code review, ensuring unit tests are adequate. You may also try fitting using these minimizers.

Fixes #16748 . 

**Release Notes** 

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by KarlPalmen** [comment]
I get ``CrystalFieldMultiSiteTest`` to fail with failure to register the ``CrystalFieldFunction``. I see this test arose in PR #20528 .

File: Framework/CurveFitting/CMakeLists.txt [hunk 1]
```diff
@@ -458,7 +444,6 @@ set ( TEST_FILES
 	MultiDomainCreatorTest.h
 	MultiDomainFunctionTest.h
 	ParameterEstimatorTest.h
-	RalNlls/NLLSTest.h
 	SpecialFunctionSupportTest.h
 )
```
Comment (KarlPalmen) [inline Framework/CurveFitting/CMakeLists.txt]: What's happened here! I don't remember making all theses changes! I just added a unit test.

**Comment by Anders-Markvardsen** [comment]
Done manual test with UnitTest data EMU00019489.nxs fitting with an older version of Manual and a version of Mantid from this branch. I get exactly the same fit when using the trust region minimizer

**Comment by Anders-Markvardsen** [comment]
Unrelated to this issue, and as a reminder for myself. Interestingly, the trust region minimizer does not appear to do well when fitting the above mentioned dataset, spectrum 1, against ExpDecayMuon from 0 to 20 or from 0.5 to 20.

**Comment by Anders-Markvardsen** [comment]
A search reveals nlls_inform has been removed from DTRS and MoreSorensen (still used by TrustRegion and TrustRegionMinimizer)

**Comment by Anders-Markvardsen** [comment]
and nlls_inform replaced with throwing exception which is OK for now. These can be modified with more soft handling as needs be, but from the test I did earlier and from the unit test none of these are thrown

**Comment by Anders-Markvardsen** [comment]
@KarlPalmen has NLLS_workspace been removed?

**Comment by KarlPalmen** [comment]
After talking with Roman we agreed that the whole of NLLS_workspace could not be removed and instead removed a significant part, so reducing the number of functions that use it.

**Review by Anders-Markvardsen — APPROVED** [review]
Function and variables names conform to coding standard, unit tests added, removed nlls_information from the two minimizers and reduced the dependence of NLLS_workspace

**Pull request merged**